e2e/test/scenarios/embedding/sdk-iframe-embedding-setup/user-settings-persistence.cy.spec.ts

```diff
@@ -0,0 +1,168 @@
+      cy.findByText("Next").click(); // Embed options step
+    });
+
+    cy.log("3. verify persisted settings are restored");
+    getEmbedSidebar().within(() => {
+      cy.findByLabelText("Allow downloads").should("be.checked");
+      cy.findByLabelText("Show chart title").should("not.be.checked");
+      cy.findByLabelText("Allow users to drill through on data points").should(
+        "be.checked",
+      );
+    });
+  });
+
+  it("persists exploration embed options", () => {
+    navigateToEmbedOptionsStep({ experience: "exploration" });
+
+    cy.log("1. set exploration settings to non-default values");
+    capturePersistSettings();
+    getEmbedSidebar().within(() => {
+      cy.findByLabelText("Allow users to save new questions")
+        .click()
+        .should("not.be.checked");
+    });
+
+    H.getIframeBody().findByText("Save").should("not.exist");
+
+    cy.log("2. reload the page");
+    waitAndReload();
+
+    getEmbedSidebar().within(() => {
+      cy.findByLabelText("Exploration").should("be.checked");
+      cy.findByText("Next").click(); // Embed options step
+
+      cy.log("3. persisted settings should be restored");
+      cy.findByLabelText("Allow users to save new questions").should(
+        "not.be.checked",
+      );
+    });
+  });
+
+  it("persists brand colors", () => {
+    navigateToEmbedOptionsStep({ experience: "dashboard" });
+
+    cy.log("1. change brand color to red");
+    cy.findByLabelText("#509EE3").click();
+
+    H.popover().within(() => {
+      capturePersistSettings();
+
+      cy.findByDisplayValue("#509EE3")
+        .should("be.visible")
+        .clear()
+        .type("rgb(255, 0, 0)")
+        .blur();
+    });
+
+    H.getIframeBody()
+      .findAllByTestId("cell-data")
+      .first()
+      .should("have.css", "color", "rgb(255, 0, 0)");
+
+    cy.log("2. reload the page");
+    waitAndReload();
+    getEmbedSidebar().within(() => {
+      cy.findByText("Next").click(); // Entity selection step
+      cy.findByText("Next").click(); // Embed options step
+    });
+
+    cy.log("3. brand color should be persisted");
+    H.getIframeBody()
+      .findAllByTestId("cell-data")
+      .first()
+      .should("have.css", "color", "rgb(255, 0, 0)");
+  });
+});
+
+const navigateToEmbedOptionsStep = ({
+  experience,
+}: {
+  experience: "dashboard" | "chart" | "exploration";
+}) => {
+  navigateToEntitySelectionStep({ experience });
+
+  getEmbedSidebar().within(() => {
+    cy.findByText("Next").click(); // Embed options step
+  });
+};
+
+// We must capture at the last embed options to change,
+// otherwise we'd miss the last PUT request.
+const capturePersistSettings = () => {
+  cy.intercept("PUT", "/api/setting/sdk-iframe-embed-setup-settings").as(
+    "persistSettings",
+  );
+};
+
+const waitAndReload = () => {
+  cy.wait("@persistSettings").then(() => {
+    // Reduce flakiness by waiting for settings to be saved.
+    cy.wait(1000);
```

Comment:
Even when waiting for the `PUT /api/setting/sdk-iframe-embed-setup-settings` request is completed and a response is returned, I found that the tests became really flaky. Waiting 1 seconds reduced the flakiness.